src/bin/rustfmt.rs

```diff
@@ -77,6 +77,19 @@ fn lookup_and_read_project_file(input_file: &Path) -> io::Result<(PathBuf, Strin
     Ok((path, toml))
 }
 
+/// read the given config file path recursively if present else read the project file path
+fn match_config_path_or_input_file(input_file: &Path,
+                                   config_path: Option<&String>)
```

Comment:
Yea, the type coercion doesn't always work as expected.. You will have to play around with this a bit. Maybe `config_path.as_ref().map(|r| &*r)` will do the trick?